cli/napi/napi_create_promise.rs

```diff
@@ -0,0 +1,17 @@
+use deno_core::napi::*;
+
+#[napi_sym::napi_sym]
+fn napi_create_promise(
+  env: napi_env,
+  deferred: *mut napi_deferred,
+  promise_out: *mut napi_value,
+) -> Result {
+  let mut env = &mut *(env as *mut Env);
+  let resolver = v8::PromiseResolver::new(env.scope).unwrap();
+  let mut global = v8::Global::new(env.scope, resolver);
+  let promise: v8::Local<v8::Value> = resolver.get_promise(env.scope).into();
+  *deferred = *(&mut global as *mut _ as *mut napi_deferred);
```

Comment:
Discussed offline with @bartlomieju and @piscisaureus 

rusty_v8 Globals are not the size of a pointer and hence the isolate handle is garbage when used for opening the global in `napi_resolve/reject_deffered`. Needs some rusty_v8 changes but is doable.

Reply:
Fixed 🎉